### PR TITLE
Move internationalization from languages to i18n

### PR DESCRIPTION
**Pull request by kghbln**
This PR is made in reference to: https://github.com/SemanticMediaWiki/SemanticMediaWiki/pull/1848#issuecomment-249402052

This PR addresses or contains:
- Moves all internationalization to the i18n directory
- Removes internationalization via php-files except for alias and magic

This PR includes:
- [ ] Tests (unit/integration)
- [x] CI build passed

Follow-up to https://github.com/SemanticMediaWiki/SemanticMediaWiki/pull/1848

### CODE REVIEW

File: includes/Setup.php [hunk 1]
```diff
@@ -176,10 +176,9 @@ private function registerI18n() {
 		$smwgIP = $this->applicationFactory->getSettings()->get( 'smwgIP' );
 
 		$this->globalVars['wgMessagesDirs']['SemanticMediaWiki'] = $smwgIP . 'i18n';
-		$this->globalVars['wgExtensionMessagesFiles']['SemanticMediaWiki'] = $smwgIP . 'languages/SMW_Messages.php';
```
Comment (kghbln) [inline includes/Setup.php]: I wonder why this worked before since there is no "SMW_Messages.php", i.e. shim. Hmm ...

File: includes/Setup.php [hunk 2]
```diff
@@ -176,10 +176,9 @@ private function registerI18n() {
...
-		$this->globalVars['wgExtensionMessagesFiles']['SemanticMediaWikiAlias'] = $smwgIP . 'languages/SMW_Aliases.php';
-		$this->globalVars['wgExtensionMessagesFiles']['SemanticMediaWikiMagic'] = $smwgIP . 'languages/SMW_Magic.php';
-		$this->globalVars['wgExtensionMessagesFiles']['SemanticMediaWikiNamespaces'] = $smwgIP . 'languages/SemanticMediaWiki.namespaces.php';
+		$this->globalVars['wgExtensionMessagesFiles']['SemanticMediaWikiAlias'] = $smwgIP . 'i18n/extra/SemanticMediaWiki.alias.php';
+		$this->globalVars['wgExtensionMessagesFiles']['SemanticMediaWikiMagic'] = $smwgIP . 'i18n/extra/SemanticMediaWiki.magic.php';
+		$this->globalVars['wgExtensionMessagesFiles']['SemanticMediaWikiNamespaces'] = $smwgIP . 'i18n/extra/SemanticMediaWiki.namespace.php';
```
Comment (kghbln) [inline includes/Setup.php]: This is something for another follow up. Namespaces are localized via the .json files and we have the "SemanticMediaWiki.namespace.php" file which appears to be superfluous.

**Comment by kghbln** [comment]
@mwjames Thanks for the tip! Let's see which kind of disaster will unfold now. Keeping fingers crossed though. Time for beddy-bye now.

**Comment by kghbln** [comment]
Yeah, this one got though. Thanks again for the pointer. The only thing that worries me is e.g. [this failiour](https://travis-ci.org/SemanticMediaWiki/SemanticMediaWiki/jobs/166001613) though it is an allowed one.

**Comment by mwjames** [comment]
> The only thing that worries me is e.g. this failiour though it is an allowed one.

This one is fine since it touches `composer.json` therefore is allowed to fail at this point because `packagist.org` has yet to get the new definition which hasn't been merged.

The one that needs further investigation is [0] as it contains `@covers` tag (uses a `SMWLanguage` class tag which is no longer valid), so removing this line from [1] should seal the deal.

```
.................................................Trying to @cover or @use not existing class or interface "\SMWLanguage"
```

PS: I don't know but maybe some documentation on smw@wiki needs an update in regards to the changed process on how to extend the `extra` language files. Further more a short description on the fallback/isLanguageRedirect language option maybe useful.

PSS: Once you feel comfortable with the new format (and after some more testing) maybe send an email to the user list and ask people with native language skills to go over the files and provide an update.

PSSS: :+1:  I'm glad that we finally can dump the `languages` folder.


[0] https://travis-ci.org/SemanticMediaWiki/SemanticMediaWiki/jobs/166001604
[1] https://github.com/SemanticMediaWiki/SemanticMediaWiki/blob/20f290cb89bce1997612719882c7239bc45a10d9/tests/phpunit/Integration/System/LanguagesAccessibilityAndIntegrityTest.php#L9

File: includes/Setup.php (hunk 1 above)
Comment (mwjames) [inline includes/Setup.php]: The depth of MW is deep.

File: includes/Setup.php (hunk 2 above)
Comment (mwjames) [inline includes/Setup.php]: > which appears to be superfluous.

Given that "... Currently[1] making namespace name translations is disabled on translatewiki.net ..." [0], I concur with you. 

[0] https://www.mediawiki.org/wiki/Localisation#Localising_namespaces_and_special_page_aliases

**Comment by mwjames** [comment]
>  @covers tag (uses a SMWLanguage class tag which is no longer valid), so removing this line from [1] should seal the deal.

Done with https://github.com/SemanticMediaWiki/SemanticMediaWiki/commit/7b4cc8a00686cd987f1b9d8e0f521f5715f10dae.

**Comment by JeroenDeDauw** [comment]
Y2.9K`arsten`

**Comment by kghbln** [comment]
> Y2.9K

? I am still a bit slow after the weekend ...

**Comment by kghbln** [comment]
> @mwjames Re https://github.com/SemanticMediaWiki/SemanticMediaWiki/pull/1896#issuecomment-252457551

> PS  I don't know but maybe some documentation on smw@wiki needs an update in regards to the changed process on how to extend the extra language files.

I guess this is undocumented. Added the label to remind me.

> PPS: Once you feel comfortable with the new format

Touching wood though it is now more consistent to the rest of the i18n - which is good.

> PPS: maybe send an email to the user list and ask people with native language skills to go over the files and provide an update.

Such mails did not even help for regular translations I am afraid. Still we should try.

**Comment by JeroenDeDauw** [comment]
You deleted 2.9k lines of code. 
mjau

**Comment by kghbln** [comment]
Ah, now I get it. :)

**Pull request merged**